kie-server-parent/kie-server-common/src/main/java/org/kie/server/common/KeyStoreHelperUtil.java

```diff
@@ -19,38 +22,44 @@
 
-    public static String loadServerPassword() {
-        String passwordKey;
-        KeyStoreHelper keyStoreHelper = new KeyStoreHelper();
-
-        try {
-            String pwdKeyAlias = System.getProperty(PROP_PWD_SERVER_ALIAS, "");
-            char[] pwdKeyPassword = System.getProperty(PROP_PWD_SERVER_PWD, "").toCharArray();
+    private static Set<String> invocationCache = new TreeSet<>();
 
-            passwordKey = keyStoreHelper.getPasswordKey(pwdKeyAlias, pwdKeyPassword);
-        } catch (RuntimeException re) {
-            logger.warn("Unable to load key store. Using password from configuration");
-            passwordKey = System.getProperty(KieServerConstants.CFG_KIE_PASSWORD, "kieserver1!");
-        }
-
-        return passwordKey;
+    public static String loadServerPassword() {
+        String defaultPassword = System.getProperty(KieServerConstants.CFG_KIE_PASSWORD, "kieserver1!");
+        return loadPasswordKey(PROP_PWD_SERVER_ALIAS, PROP_PWD_SERVER_PWD, defaultPassword);
     }
 
     public static String loadControllerPassword(final KieServerConfig config) {
         return loadControllerPassword(config.getConfigItemValue(KieServerConstants.CFG_KIE_CONTROLLER_PASSWORD, "kieserver1!"));
     }
 
     public static String loadControllerPassword(final String defaultPassword) {
-        KeyStoreHelper keyStoreHelper = new KeyStoreHelper();
+        return loadPasswordKey(PROP_PWD_CTRL_ALIAS, PROP_PWD_CTRL_PWD, defaultPassword);
+    }
 
+    private static String loadPasswordKey(String pwdKeyAliasProperty, String pwdKeyPasswordProperty, String defaultPassword) {
+        String passwordKey;
+        KeyStoreHelper keyStoreHelper = new KeyStoreHelper();
         try {
-            String pwdKeyAlias = System.getProperty(PROP_PWD_CTRL_ALIAS, "");
-            char[] pwdKeyPassword = System.getProperty(PROP_PWD_CTRL_PWD, "").toCharArray();
-
-            return keyStoreHelper.getPasswordKey(pwdKeyAlias, pwdKeyPassword);
+            String pwdKeyAlias = System.getProperty(pwdKeyAliasProperty, "");
+            String pwdKeyPassword = System.getProperty(pwdKeyPasswordProperty, "");
+            passwordKey = keyStoreHelper.getPasswordKey(pwdKeyAlias, pwdKeyPassword.toCharArray());
         } catch (RuntimeException re) {
-            logger.warn("Unable to load key store. Using password from configuration");
-            return defaultPassword;
+            passwordKey = defaultPassword;
+            if (hasNotBeenInvoked(pwdKeyAliasProperty, pwdKeyPasswordProperty, passwordKey)) {
+                // only print when it has not been invoked
+                logger.warn("Unable to load key store. Using password from configuration");
+            }
+        }
+        return passwordKey;
+    }
+
+    private static boolean hasNotBeenInvoked(String pwdKeyAliasProperty, String pwdKeyPasswordProperty, String passwordKey) {
+        final String key = pwdKeyAliasProperty + "-" + pwdKeyPasswordProperty;
```

Comment:
I just noticed that we concatenate 2 properties, but do we need them both? Maybe `pwdKeyAliasProperty` would be enough to distinguish between server and controller keys?

Reply:
In theory yes, but I don't presume anything.